code/game/gamemodes/dynamic/dynamic.dm

```diff
@@ -282,340 +285,264 @@ GLOBAL_VAR_INIT(dynamic_forced_storyteller, null)
-				return FALSE
-			starting_rule = pickweight(drafted_rules)
-
-	log_game("DYNAMIC: Picked a ruleset: [starting_rule.name]")
-
-	roundstart_rules -= starting_rule
-	drafted_rules -= starting_rule
-
-	starting_rule.trim_candidates()
-	if (starting_rule.pre_execute())
-		log_threat("[starting_rule.ruletype] - <b>[starting_rule.name]</b> [starting_rule.cost + starting_rule.scaled_times * starting_rule.scaling_cost] threat", verbose = TRUE)
-		if(starting_rule.flags & HIGHLANDER_RULESET)
-			highlander_executed = TRUE
-		else if(starting_rule.flags & ONLY_RULESET)
+		round_start_budget_left -= cost
+
+		rulesets_picked[ruleset] += 1
+
+		if (ruleset.flags & HIGH_IMPACT_RULESET)
+			for (var/_other_ruleset in drafted_rules)
+				var/datum/dynamic_ruleset/other_ruleset = _other_ruleset
+				if (other_ruleset.flags & HIGH_IMPACT_RULESET)
+					drafted_rules[other_ruleset] = null
+
+		if (ruleset.flags & LONE_RULESET)
+			drafted_rules[ruleset] = null
+
+	for (var/ruleset in rulesets_picked)
+		spend_roundstart_budget(picking_roundstart_rule(ruleset, rulesets_picked[ruleset] - 1))
+
+	update_log()
+
+/// Initializes the round start ruleset provided to it. Returns how much threat to spend.
+/datum/game_mode/dynamic/proc/picking_roundstart_rule(datum/dynamic_ruleset/roundstart/ruleset, scaled_times = 0, forced = FALSE)
+	log_game("DYNAMIC: Picked a ruleset: [ruleset.name], scaled [scaled_times] times")
+
+	ruleset.trim_candidates()
+	var/added_threat = ruleset.scale_up(roundstart_pop_ready, scaled_times)
+
+	if(ruleset.pre_execute(roundstart_pop_ready))
+		threat_log += "[worldtime2text()]: Roundstart [ruleset.name] spent [ruleset.cost + added_threat]. [ruleset.scaling_cost ? "Scaled up [ruleset.scaled_times]/[scaled_times] times." : ""]"
+		if(ruleset.flags & ONLY_RULESET)
 			only_ruleset_executed = TRUE
-		executed_rules += starting_rule
-		for(var/datum/dynamic_ruleset/roundstart/rule in drafted_rules)
-			if(check_blocking(rule.blocking_rules, executed_rules))
-				drafted_rules -= rule
-			if(highlander_executed && rule.flags & HIGHLANDER_RULESET)
-				drafted_rules -= rule
-			if(!rule.ready())
-				drafted_rules -= rule // And removing rules that are no longer eligible
-		return TRUE
+		if(ruleset.flags & HIGH_IMPACT_RULESET)
+			high_impact_ruleset_executed = TRUE
+		executed_rules += ruleset
+		return ruleset.cost + added_threat
 	else
-		stack_trace("The starting rule \"[starting_rule.name]\" failed to pre_execute.")
-	return FALSE
+		stack_trace("The starting rule \"[ruleset.name]\" failed to pre_execute.")
+	return 0
 
 /// Mainly here to facilitate delayed rulesets. All roundstart rulesets are executed with a timered callback to this proc.
 /datum/game_mode/dynamic/proc/execute_roundstart_rule(sent_rule)
 	var/datum/dynamic_ruleset/rule = sent_rule
 	if(rule.execute())
-		current_rules += rule
-		SSblackbox.record_feedback("associative","dynamic_rulesets",1,rule.get_blackbox_info())
```

Comment:
technically this one should be there but if you removed rule.get_blackbox_info() don't bother

Reply:
It's gone